src/lib/components/common/Dropdown.svelte

```diff
@@ -0,0 +1,170 @@
+<script lang="ts" context="module">
+  export type { Placement } from "@floating-ui/dom";
+</script>
+
+<script lang="ts">
+  import { onMount } from "svelte";
+  import { writable } from "svelte/store";
+
+  import {
+    autoUpdate,
+    computePosition,
+    flip,
+    offset as offsetFn,
+    shift,
+    type Placement,
+  } from "@floating-ui/dom";
+
+  export let position: Placement = "bottom-start";
+  export let offset: number = 4;
+  export let overlay = false;
+  export let border = false;
+
+  let dropdown: HTMLDivElement;
+  let anchor: HTMLDivElement;
+  let cleanup: () => void;
+
+  const dropdownCoords = writable({ x: 0, y: 0 });
+
+  $: isOpen = dropdown?.style.display === "block";
+
+  function update() {
+    const middleware = [offsetFn(offset), shift({ padding: 6 }), flip()];
+    computePosition(anchor, dropdown, {
+      placement: position,
+      middleware,
+    }).then(({ x, y }) => {
+      dropdownCoords.set({ x, y });
+    });
+  }
+
+  function open() {
+    dropdown.style.display = "block";
+    update();
+    cleanup = autoUpdate(anchor, dropdown, update);
+  }
+
+  function close() {
+    dropdown.style.display = "";
+    cleanup?.();
+  }
+
+  function toggle() {
+    if (isOpen) {
+      close();
+    } else {
+      open();
+    }
+  }
+
+  // Toggle the dropdown when the anchor is interacted with
+  function toggleOnAnchorEvent(event: MouseEvent | KeyboardEvent) {
+    switch (event.type) {
+      case "click":
+        toggle();
+        break;
+      case "keydown":
+        const key = (event as KeyboardEvent).key;
+        if (["Spacebar", " ", "Enter", "ArrowDown"].includes(key)) {
+          toggle();
+        }
+        break;
+      default:
+    }
+  }
+
+  // Close the dropdown when a click or escape is made outside its subtree
+  function closeOnEventOutside(event: MouseEvent) {
+    if (event.target instanceof Element) {
+      const outside = !dropdown.contains(event.target);
+      const notTitle = !anchor.contains(event.target);
+      if (outside && notTitle) {
+        close();
+      }
+    }
+  }
+
+  // Close the dropdown when using the Escape key
+  function closeOnEscape(event: KeyboardEvent) {
+    const key = (event as KeyboardEvent).key;
+    if (key === "Escape") {
+      close();
+    }
+  }
+
+  onMount(() => {
+    document.addEventListener("click", closeOnEventOutside);
+    document.addEventListener("keydown", closeOnEscape);
+
+    return () => {
+      document.removeEventListener("click", closeOnEventOutside);
+      document.removeEventListener("keydown", closeOnEscape);
+    };
+  });
+</script>
```

Comment:
At some point we should move these events out of lifecycle and into `<svelte:document>` event handlers, but that doesn't have to happen in this PR.

Reply:
Makes sense to roll this into a keyboard shortcut task and mention this in the issue